src/puppetlabs/trapperkeeper/services/status/ringutils.clj

```diff
@@ -0,0 +1,65 @@
+;; TODO: Much of this code is copied from other projects. It should
+;; probably be moved into a shared library or some such.
```

Comment:
Ha, yeah ... mess proliferation.  Although it probably needs some cleanup before landing in ring-middleware.